tools/cli/internal/openapi/filter/hidden_envs.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package filter
+
+import (
+	"log"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+)
+
+const (
+	hiddenEnvsExtensionName     = "x-xgen-hidden-env"
+	hiddenEnvsExtionsionKeyName = "envs"
+)
+
+type HiddenEnvsFilter struct{}
+
+func (f *HiddenEnvsFilter) Apply(oas *openapi3.T, metadata *Metadata) error {
+	for _, pathItem := range oas.Paths.Map() {
+		if err := applyOnPath(pathItem, metadata); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
same question as https://github.com/mongodb/openapi/pull/72#discussion_r1678973829 

And I use this thread to answer @blva I don't have the context on the usage of the struct but I think it would be both for th both of you to sync on the design you want for your lib, do you really need this struct and store a state or can the methods be really stateless?

Reply:
updated based on #72 (comment)